paymentsheet/src/test/java/com/stripe/android/paymentelement/confirmation/link/LinkConfirmationActivityTest.kt

```diff
@@ -0,0 +1,248 @@
+package com.stripe.android.paymentelement.confirmation.link
+
+import android.app.Activity
+import android.app.Application
+import android.app.Instrumentation
+import android.content.Intent
+import android.net.Uri
+import android.util.Base64
+import androidx.core.os.bundleOf
+import androidx.test.core.app.ActivityScenario
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.espresso.intent.Intents.intended
+import androidx.test.espresso.intent.Intents.intending
+import androidx.test.espresso.intent.matcher.IntentMatchers.hasComponent
+import androidx.test.espresso.intent.rule.IntentsRule
+import app.cash.turbine.test
+import com.google.common.truth.Truth.assertThat
+import com.stripe.android.core.utils.FeatureFlags
+import com.stripe.android.link.LinkConfiguration
+import com.stripe.android.paymentelement.confirmation.ConfirmationHandler
+import com.stripe.android.paymentelement.confirmation.PaymentElementConfirmationTestActivity
+import com.stripe.android.paymentelement.confirmation.PaymentMethodConfirmationOption
+import com.stripe.android.paymentelement.confirmation.assertComplete
+import com.stripe.android.paymentelement.confirmation.assertConfirming
+import com.stripe.android.paymentelement.confirmation.assertIdle
+import com.stripe.android.paymentelement.confirmation.assertSucceeded
+import com.stripe.android.payments.paymentlauncher.InternalPaymentResult
+import com.stripe.android.paymentsheet.createTestActivityRule
+import com.stripe.android.paymentsheet.state.PaymentElementLoader
+import com.stripe.android.paymentsheet.verticalmode.ShampooRule
+import com.stripe.android.testing.FeatureFlagTestRule
+import com.stripe.android.testing.PaymentConfigurationTestRule
+import com.stripe.android.testing.PaymentIntentFactory
+import com.stripe.android.testing.PaymentMethodFactory
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.test.UnconfinedTestDispatcher
+import kotlinx.coroutines.test.runTest
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.ParameterizedRobolectricTestRunner
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+
+@RunWith(ParameterizedRobolectricTestRunner::class)
+internal class LinkConfirmationActivityTest(private val nativeLinkEnabled: Boolean) {
+    private val application = ApplicationProvider.getApplicationContext<Application>()
+
+    @get:Rule
+    val intentsRule = IntentsRule()
+
+    @get:Rule
+    val testActivityRule = createTestActivityRule<PaymentElementConfirmationTestActivity>()
+
+    @get:Rule
+    val paymentConfigurationTestRule = PaymentConfigurationTestRule(
+        context = application,
+    )
+
+    @get:Rule
+    val featureFlagTestRule = FeatureFlagTestRule(
+        featureFlag = FeatureFlags.nativeLinkEnabled,
+        isEnabled = false
+    )
+
+    @get:Rule
+    val shampooRule = ShampooRule(200)
```

Comment:
🙈

Reply:
Is it possible to make a lint rule to avoid this?

Reply:
Yeah, we can look for the specific `ShampooRule` import and produce a lint error from it.